src/engine/renderer/tr_shader.cpp

```diff
@@ -3985,49 +4116,56 @@ static bool ParseShader( const char *_text )
 		// diffuseMap <image>
 		else if ( !Q_stricmp( token, "diffuseMap" ) )
 		{
-			ParseDiffuseMap( &stages[ s ], text );
+			Log::Warn("deprecated standalone stage '%s' in shader '%s'", token, shader.name );
+			ParseDiffuseMap( &stages[ s ], text, TB_COLORMAP );
 			s++;
 			continue;
 		}
 		// normalMap <image>
 		else if ( !Q_stricmp( token, "normalMap" ) || !Q_stricmp( token, "bumpMap" ) )
 		{
-			ParseNormalMap( &stages[ s ], text );
+			Log::Warn("deprecated standalone stage '%s' in shader '%s'", token, shader.name );
+			ParseNormalMap( &stages[ s ], text, TB_COLORMAP );
 			s++;
 			continue;
 		}
 		// specularMap <image>
 		else if ( !Q_stricmp( token, "specularMap" ) )
 		{
-			ParseSpecularMap( &stages[ s ], text );
+			Log::Warn("deprecated standalone stage '%s' in shader '%s'", token, shader.name );
```

Comment:
Could you put a link or some kind of clue for users to learn what the alternative syntax is?

Reply:
I'll do

Reply:
I mean, clue in code or in warning text?

Reply:
warning text